### PR TITLE
Adding gradient accumulation and random cropping

### PR DESCRIPTION
**Pull request by arslan-chaudhry**
Summary: This commit adds gradient accumulation strategy where
the gardients are accumulated for 'X' number of steps before
updating the weights, effectively mimicing the "iter_size"
functionality of Caffe. This allows for arbitrary length batch
sizes on the GPUs with less memory. Also, random cropping and mirroring
is added as means for better data augmentation.

Changes to be committed:
	modified:   README.md
	modified:   deeplab_resnet/image_reader.py
	modified:   evaluate.py
	modified:   fine_tune.py
	modified:   train.py
	modified:   train_msc.py

Signed-off-by: Arslan Chaudhry (arslan_mac@yahoo.com)

### CODE REVIEW

**Comment by DrSleep** [comment]
@arslan-chaudhry, I have made small changes: most notably, during the gradients accumulation, the gradients are divided by the number of gradient steps immediately without the conversion to numpy arrays.
